neurodamus/utils/multimap.py

```diff
@@ -7,24 +7,24 @@
 import numpy as np
 
 
-class MultiMap(collections.abc.Mapping):
-    """A memory-efficient map, which accepts duplicates"""
+class GroupedMultiMap(collections.abc.Mapping):
+    """A Multimap which groups values by key in a list."""
 
-    __slots__ = ("_keys", "_values")
+    __slots__ = (
+        "_keys",  # stored as `np.array`
+        "_values",  # stored as `list`
+    )
 
-    def __init__(self, np_keys, values, presorted=False):
-        """Constructor for OrderedMap
+    def __init__(self, np_keys, values):
+        """Init GroupedMultiMap.
 
         Args:
             np_keys: The numpy array of the keys. Can be empty
             values: The array of the values, can be any indexable, but better if numpy
         """
         assert len(np_keys) == len(values), "Keys and values must have the same length"
-        if presorted:
-            self._keys = np_keys
-            self._values = values
-        else:
-            self._keys, self._values = self.sort_together(np_keys, values)
+        self._keys, self._values = self.sort_together(np_keys, values)
+        self._keys, self._values = self._duplicates_to_list(self._keys, self._values)
```

Comment:
while you are at that I think duplicates_to_list may be made more pythonic:

```
from itertools import groupby

@staticmethod
def _duplicates_to_list(np_keys, values):
    grouped_keys = []
    grouped_values = []

    for key, group in groupby(zip(np_keys, values), key=lambda x: x[0]):
        grouped_keys.append(key)
        grouped_values.append([val for _, val in group])

    return np.array(grouped_keys), grouped_values
```

Reply:
also docstrings for sort_together and _duplicates_to_list. It is not immediately clear what they are supposed to do